pinot-common/src/test/java/com/linkedin/pinot/common/utils/TarGzCompressionUtilsTest.java

```diff
@@ -122,4 +128,61 @@ public void testEmptyDirectory()
     Assert.assertEquals(segmentFiles.length, 0);
 
   }
+
+  @Test
+  public void testFailurePath() throws IOException, ArchiveException {
+    File metaFile = new File(segmentDir, "metadata.properties");
+    metaFile.createNewFile();
+
+    // We guarantee metaFile isn't a directory.
+    Assert.assertFalse(metaFile.isDirectory());
+
+    File tarGzPath = new File(tarDir, SEGMENT_NAME + ".tar.gz");
+    mockIncorrectData(metaFile, tarGzPath);
+
+    try {
+      TarGzCompressionUtils.unTar(tarGzPath, untarDir);
+      Assert.fail("Did not get exception!!");
+    } catch (IOException e) {
+      Assert.assertTrue(e.getMessage().contains("Tar file must not be untarred outside of the target output directory!"));
```

Comment:
If you are checking for exact msg in the exception, then please define a public static string in the code and re-use it here.